earthfile2llb/converter.go

```diff
@@ -1197,6 +1216,12 @@ func (c *Converter) FinalizeStates(ctx context.Context) (*states.MultiTarget, er
 		// Should never happen.
 		return nil, errors.New("internal error: stack not at base in FinalizeStates")
 	}
+
+	if len(c.persistentCacheDirs) > 0 {
```

Comment:
nit: if guard seems unnessisary as the function does a `for _, cacheDir := range c.persistentCacheDirs {` which doesn't seem to have any side-effects when it's empty. I think simply calling `c.persistCacheVolumes()` would lead to less code-bloat.

Reply:
💯 agree